apps/event-worker/src/resource-scan/google/vpc.ts

```diff
@@ -16,61 +17,117 @@ import { getGoogleClient } from "./client.js";
-            "Google Console": `https://console.cloud.google.com/networking/networks/details/${network.name}?project=${project}`,
-          }),
-          "google/project": project,
-          "google/self-link": network.selfLink,
-          "google/creation-timestamp": network.creationTimestamp,
-          "google/description": network.description,
-          ...network.peerings?.reduce(
-            (acc, peering) => ({
-              ...acc,
-              [`google/peering/${peering.name}`]: JSON.stringify({
-                network: peering.network,
-                state: peering.state,
-                autoCreateRoutes: peering.autoCreateRoutes,
-              }),
-            }),
-            {},
-          ),
-        }),
+        name,
+        region,
+        gatewayAddress: subnet.gatewayAddress ?? "",
+        cidr: subnet.ipCidrRange ?? "",
+        type: subnet.purpose === "INTERNAL" ? "internal" : "external",
+        secondaryCidrs: subnet.secondaryIpRanges?.map((r) => ({
+          name: r.rangeName ?? "",
+          cidr: r.ipCidrRange ?? "",
+        })),
       };
     });
+};
+
+const getNetworkResources = async (
+  clients: { networksClient: NetworksClient; subnetsClient: SubnetworksClient },
+  project: string,
+  networks: google.cloud.compute.v1.INetwork[],
+): Promise<CloudVPCV1[]> =>
+  await Promise.all(
+    networks
+      .filter((n) => n.name != null)
+      .map(async (network) => {
+        const subnets = await Promise.all(
+          (network.subnetworks ?? []).map((subnet) =>
+            getSubnetDetails(clients.subnetsClient, project, subnet),
+          ),
+        );
+        return {
+          name: network.name!,
+          identifier: `${project}/${network.name}`,
+          version: "cloud/v1",
+          kind: "VPC",
+          config: {
+            name: network.name!,
+            provider: "google",
+            region: "global",
+            project,
+            cidr: network.IPv4Range ?? undefined,
+            mtu: network.mtu ?? undefined,
+            subnets: subnets.filter(isPresent),
+          },
+          metadata: omitNullUndefined({
+            [ReservedMetadataKey.ExternalId]: network.id?.toString(),
+            [ReservedMetadataKey.Links]: JSON.stringify({
+              "Google Console": `https://console.cloud.google.com/networking/networks/details/${network.name}?project=${project}`,
+            }),
+            "google/project": project,
+            "google/self-link": network.selfLink,
+            "google/creation-timestamp": network.creationTimestamp,
+            "google/description": network.description,
+            ...(network.peerings?.reduce(
+              (acc, peering) => {
+                acc[`google/peering/${peering.name}`] = JSON.stringify({
+                  network: peering.network,
+                  state: peering.state,
+                  autoCreateRoutes: peering.autoCreateRoutes,
+                });
+                return acc;
+              },
```

Comment:
is there a way to lodash this to make it more readable using the fluent api interface